### ISSUE TITLE
Edit over SSH

### ISSUE THREAD

**Issue by firelizzard18** [Microsoft/vscode#44378]
### Issue Type
Feature Request

### Description

I would like an action, "Open Remote File (SFTP)", that would open a file on a remote system, using SFTP (SSH FTP).

### This would

* Open an editor pane with the contents of the file
* Update the file on the remote system whenever I save the editor pane

### This would not

* Show anything in the explorer pane or any other side pane, beyond the editor entry in open editors

### Why this is not a duplicate?

Technically, this is a duplicate of #23076. However, that issue was incorrectly closed as a duplicate of allow reading from stdin (#7320/#6161). This was incorrect as "remote editing by SSH" is very different from "allow reading from stdin".

This is distinct from #27423 as that issue is "open remote folder via SSH", not _file_. From a UI perspective, opening a file is much simpler, as the only UI needed is an editor pane and a few commands. A remote folder requires the contents of that folder to be displayed somewhere, such as the explorer pane.

### VS Code Info

VS Code version: Code 1.20.1 (f88bbf9137d24d36d968ea6b2911786bfe103002, 2018-02-13T15:34:36.336Z)
OS version: Windows_NT x64 10.0.16299

**Comment by jrieken** [comment]
Kind of a dupe of #29194 but once that (monster) item is done extension will be able to implement this. There is already a experimental implementation for ftp doing this (see https://github.com/jrieken/ftp-sample). So, ultimately closing as 'extension-candidate' but the API hasn't been finalised yet.

**Comment by firelizzard18** [comment]
@jrieken Does the same hold true for #27423? That is, being closed as an extension candidate pending the remote API? Because it sounds like the report API is a remote FS API, which sounds like it would support directories.

**Comment by jrieken** [comment]
> report API is a remote FS API, which sounds like it would support directories.

Yup, that's what it will be

**Comment by mjbvz** [comment]
We just announced remote development with VS Code, check out the [blog post](https://code.visualstudio.com/blogs/2019/05/02/remote-development) for details
